src/source/Common/Curl/CurlCall.c

```diff
@@ -55,6 +55,19 @@ STATUS blockingCurlCall(PRequestInfo pRequestInfo, PCallInfo pCallInfo)
         curl_easy_setopt(curl, CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2);
     }
 
+    // setup proxy using environment variable
+    if(getenv("CURLOPT_PROXY") != NULL) {
+        curl_easy_setopt(curl, CURLOPT_PROXY, getenv("CURLOPT_PROXY"));
+    }
+
+    if(getenv("CURLOPT_PROXYUSERNAME") != NULL) {
+        curl_easy_setopt(curl, CURLOPT_PROXYUSERNAME, getenv("CURLOPT_PROXYUSERNAME"));
+    }
+
+    if(getenv("CURLOPT_PROXYPASSWORD") != NULL) {
+        curl_easy_setopt(curl, CURLOPT_PROXYPASSWORD, getenv("CURLOPT_PROXYPASSWORD"));
+    }
+
```

Comment:
Let's not hard code these, how about adding #define's here:

https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/source/Common/Curl/CurlCall.h#L13